crates/biome_js_analyze/src/lint/nursery/use_throw_new_error.rs

```diff
@@ -0,0 +1,180 @@
+use std::ops::Not;
+
+use biome_analyze::{
+    context::RuleContext, declare_rule, ActionCategory, Ast, FixKind, Rule, RuleDiagnostic,
+    RuleSource,
+};
+use biome_console::markup;
+use biome_diagnostics::Applicability;
+use biome_js_factory::make;
+use biome_js_syntax::{
+    AnyJsExpression, JsCallExpression, JsNewExpression, JsParenthesizedExpression, JsSyntaxKind, T,
+};
+use biome_rowan::{AstNode, BatchMutationExt, TriviaPieceKind};
+use lazy_static::lazy_static;
+use regex::Regex;
+
+use crate::JsRuleAction;
+
+declare_rule! {
+    /// Require new when throwing an error
+    ///
+    /// While it's possible to create a new error without using the new keyword, it's better to be explicit.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// throw Error();
+    /// ```
+    /// ```js,expect_diagnostic
+    /// throw TypeError('biome');
+    /// ```
+    /// ```js,expect_diagnostic
+    /// throw lib.TypeError();
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// throw new Error();
+    /// ```
+    /// ```js
+    /// throw new TypeError('biome');
+    /// ```
+    /// ```js
+    /// throw new lib.TypeError();
+    /// ```
+    ///
+    pub UseThrowNewError {
+        version: "next",
+        name: "useThrowNewError",
+        sources: &[RuleSource::EslintUnicorn("throw-new-error")],
+        recommended: false,
+        fix_kind: FixKind::Unsafe,
+    }
+}
+
+impl Rule for UseThrowNewError {
+    type Query = Ast<JsCallExpression>;
+    type State = ();
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+
+        if is_in_throw_statement(node.clone()).not() {
+            return None;
+        }
+
+        let callee = &node.callee().ok()?.omit_parentheses();
+        let name = match callee {
+            AnyJsExpression::JsIdentifierExpression(ident_expr) => Some(ident_expr.text()),
+            AnyJsExpression::JsStaticMemberExpression(member_expr) => {
+                let member_expr = member_expr.member().ok()?.text();
+
+                Some(member_expr)
+            }
```

Comment:
We could avoid the String allocation of `text()` by retrieving a text token:
```suggestion
            AnyJsExpression::JsIdentifierExpression(ident_expr) => ident_expr.name()>ok()?.name().ok(),
            AnyJsExpression::JsStaticMemberExpression(member_expr) => {
                Some(member_expr.member().ok()?.value_token().ok()?.token_text_trimmed())
            }
```